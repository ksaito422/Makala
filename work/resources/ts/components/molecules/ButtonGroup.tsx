import React from 'react';
import { Button } from '../atoms/Button';

type Props = {
  disabledCard: boolean,
  disabledPreview: boolean,
  CardOnClick: () => void,
  PreviewOnClick: () => void,
}

export const ButtonGroup: React.FC<Props> = (props: Props) => {
  return (
    <>
      <Button
        disabled={props.disabledCard}
        onClick={props.CardOnClick}
      >
        CARD
      </Button>
      <Button
        disabled={props.disabledPreview}
        onClick={props.PreviewOnClick}
      >
        PREVIEW
      </Button>
    </>
  )
}
