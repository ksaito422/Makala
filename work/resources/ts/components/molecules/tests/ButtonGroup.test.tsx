import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { ButtonGroup } from '../ButtonGroup';

// メモリリークを避けるために、各テスト後にすべてのコンテンツを消去する
afterEach(cleanup);

describe('ButtonGroup', () => {
  it('スナップショットテスト', () => {
    const props = {
      disabledCard: false,
      disabledPreview: false,
      cardOnClick: jest.fn,
      previewOnClick: jest.fn,
    };

    const { baseElement } = render(<ButtonGroup {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
