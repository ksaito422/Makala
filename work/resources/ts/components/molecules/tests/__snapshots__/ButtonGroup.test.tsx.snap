// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup スナップショットテスト 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        CARD
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        PREVIEW
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
